File: Qubit Server/dev/src/org/labrad/qubits/resources/MicrowaveSource.java
package org.labrad.qubits.resources;

import java.util.Set;

import com.google.common.collect.Sets;

public class MicrowaveSource implements Resource {
  String name;
  String server;
  String device;

  Set<MicrowaveBoard> boards = Sets.newHashSet();

  public static MicrowaveSource create(String name) {
    MicrowaveSource board = new MicrowaveSource(name);
    return board;
  }

  public MicrowaveSource(String name) {
    this.name = name;
    this.device = name;
  }

  public void addMicrowaveBoard(MicrowaveBoard board) {
    boards.add(board);
  }

  public Set<MicrowaveBoard> getMicrowaveBoards() {
    return boards;
  }

  public String getName() {
    return name;
  }

  public String getServer() {
    return server;
  }

  public String getDevice() {
    return device;
  }
}

File: QubitServer/src/main/java/org/labrad/qubits/resources/MicrowaveSource.scala
package org.labrad.qubits.resources

import java.util.Set

import com.google.common.collect.Sets

object MicrowaveSource {
  def create(name: String): MicrowaveSource = {
    new MicrowaveSource(name)
  }
}

class MicrowaveSource(val name: String) extends Resource {
  private val boards: Set[MicrowaveBoard] = Sets.newHashSet()

  def addMicrowaveBoard(board: MicrowaveBoard): Unit = {
    boards.add(board)
  }

  def getMicrowaveBoards(): Set[MicrowaveBoard] = {
    boards
  }

  def getName(): String = name
}
